### PR TITLE
Add script for upgrading target release

### PR DESCRIPTION
**Pull request by zacatac**
See script for usage instructions.

### CODE REVIEW

**Comment by zacatac** [comment]
cc: @imrehg

**Comment by thgreasi** [comment]
Thanks for your contribution on this.
We will review shortly :+1:

File: pin-devices-targeting-release.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+## This script sets devices targeting a given commit to target a new commit
+## Usage: ./pin-devices-to-target-release <FULL_COMMIT_HASH> <NEW_FULL_COMMIT_HASH>
```
Comment (thgreasi) [inline pin-devices-targeting-release.sh]: Would it make more sense if we used a slightly different name like `./update-pinned-devices` or `./update-devices-pinned-to-release`?
My only point is that `./pin-devices-to-target-release` doesn't describe that the target group is a set of already pinned devices.

File: pin-devices-targeting-release.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    echo "Please specify a commit to upgrade from"
+    exit 1
+elif [ -z $NEW_COMMIT ]; then
+    echo "Please specify a commit to upgrade to"
+    exit 1
+else
+    RELEASE_ID=$(./get-release-id.sh $COMMIT)
+    NEW_RELEASE_ID=$(./get-release-id.sh $NEW_COMMIT)
+fi
+echo "setting devices with commit $COMMIT with release = $RELEASE_ID to $NEW_COMMIT with release = $NEW_RELEASE_ID"
```
Comment (thgreasi) [inline pin-devices-targeting-release.sh]: While testing this, I realized that we need some guards against invalid input. Specifically
* if the source release isn't found, then the source release id that will be used is `null`, which means that a typo in the terminal could pin all unpinned devices of the app.
* the target release has the same issue, so a typo in the target release hash would unpin all devices running the specified commit.

EG:
```
./pin-devices-targeting-release.sh 9af01a00ac47cca0ed3c3bb316644f3be6eb5774 typocommit
# setting devices with commit 9af01a00ac47cca0ed3c3bb316644f3be6eb5774 with release = 985483 to typo with release = null
```
which means that all devices pinned on 9af01a00ac47cca0ed3c3bb316644f3be6eb5774 will suddenly get unpinned and update to the latest release of the app.

For the above two cases I think that we should add two checks (one for the `RELEASE_ID` and one for the `NEW_RELEASE_ID`), which would state that the script was unable to find the provided source or target release and then exit.

**Comment by thgreasi** [comment]
Oh I almost forgot to mention that in order to get the `ResinCI/check-versionist/versionist` check to pass, I will need you to amend the commit message to also include `Change-type: minor` in the description of your commit.
That's required since at Balena we follow some commit guidelines so that our CI system can reason about our PRs.
Here is an example of how the commit message could look like
```
Add script for upgrading target release

Change-type: minor
```
and here is an example from an older commit in this repo
See: https://github.com/balena-io-projects/staged-releases/pull/26/commits/d7b3f989618bd6f4a7ffd8a1776d6f7a8baff3b4

Once again, thanks for your contribution on this and give us a note if you need any help.

**Comment by zacatac** [comment]
Hey @thgreasi thanks for the review. I'll try to make some time this weekend to make the changes you requested.

**Comment by thgreasi** [comment]
Hey @zacatac any updates on this?
Give me a note if I can help you with anything or if you prefer to pick it up for you from here.

**Comment by thgreasi** [comment]
Hey @zacatac any updates on this?
Would it work better for you if someone took this PR over to get it merged (while preserving your original commit)?

**Comment by zacatac** [comment]
@thgreasi Sorry about the crazy delay on this. Feel free to reopen and takeover if you want.

**Pull request closed without merging**